File: leetcode-js/356-line-reflection.js
/**
 * @param {number[][]} points
 * @return {boolean}
 */
var isReflected = function(points) {
    if (!points.length) {
        return true;
    }

    const Xs = points.map(([x]) => x);
    const Xmin = Math.min(...Xs);
    const Xmax = Math.max(...Xs);

    const set = new Set(points.map(([x, y]) => `${x},${y}`));

    for (const [x, y] of points) {
        if (!set.has(`${Xmax + Xmin - x},${y}`)) {
            return false;
        }
    }

    return true;
};

console.log(isReflected([[1,2],[2,2],[1,4],[2,4]]));

File: leetcode-java/356-line-reflection.java
public class Solution {
    public boolean isReflected(int[][] points) {
        Set<Integer> set = new HashSet<>();
        int maxX = Integer.MIN_VALUE;
        int minX = Integer.MAX_VALUE;

        for (int[] point : points) {
            maxX = Math.max(maxX, point[0]);
            minX = Math.min(minX, point[0]);
            set.add(Arrays.hashCode(point));
        }

        int sum = maxX + minX;

        for (int[] point : points) {
            if (!set.contains(Arrays.hashCode(new int[]{sum - point[0], point[1]}))) {
                return false;
            }
        }

        return true;
    }
}
